pkg/mapper/dynamicfile/mapper.go

```diff
@@ -66,7 +67,8 @@ func (m *DynamicFileMapper) match(token *token.Identity, mappedARN, mappedUserID
 		// If ARN is provided, ARN must be validated along with UserID.  This avoids having to
 		// support IAM user name/ARN changes. Without preventing this the mapping would look
 		// invalid but still work and auditing would be difficult/impossible.
-		if mappedARN != "" && token.ARN != mappedARN {
+		strippedArn, _ := arn.StripPath(mappedARN)
```

Comment:
Can you add a `// TODO:` that says something along the lines of: this should potentially be moved to canonicalize or somehow shared by other mappers because it could fix the path issue but needs to be tested with other mappers first, e.g. in a cluster with existing custom resources for the CRD mapper.